psi4/src/psi4/lib3index/dfhelper.cc

```diff
@@ -604,31 +611,48 @@ void DFHelper::prepare_AO_wK_core() {
         size_t begin = pshell_aggs_[start];
         size_t end = pshell_aggs_[stop + 1] - 1;
 
-        // compute (A|mn)
-        timer_on("DFH: AO Construction");
-        compute_sparse_pQq_blocking_p_symm(start, stop, M1p, eri);
-        timer_off("DFH: AO Construction");
-
-        // contract full metric inverse
-        timer_on("DFH: AO-Met. Contraction");
-        contract_metric_AO_core_symm(M1p, m1ppq, met1p, begin, end);
-        timer_off("DFH: AO-Met. Contraction");
-
-        // contract half metric inverse
-        timer_on("DFH: AO-Met. Contraction");
-        contract_metric_AO_core_symm(M1p, ppq, metp, begin, end);
-        timer_off("DFH: AO-Met. Contraction");
-
-        // compute (A|w|mn)
-        timer_on("DFH: wAO Construction");
-        compute_sparse_pQq_blocking_p_symm(start, stop, M1p, weri);
-        timer_off("DFH: wAO Construction");
+        if ( wcombine_ ) {
+            // computes (Q|mn) and (Q|w|mn)
+            timer_on("DFH: AO Construction");
+            compute_sparse_pQq_blocking_p_symm_abw(start,stop, M1p, M2p, eri, weri);
+            timer_off("DFH: AO Construction");
+    
+            // contract half metric inverse I'm planning to get rid of this
+            /* timer_on("DFH: AO-Met. Contraction");
+            contract_metric_AO_core_symm(M1p, ppq, metp, begin, end);
+            timer_off("DFH: AO-Met. Contraction"); */
+    
+            // computes  [J^{-1.0}](Q|mn)
+            timer_on("DFH: AO-Met. Contraction");
+            contract_metric_AO_core_symm(M1p, m1ppq, met1p, begin, end);
+            timer_off("DFH: AO-Met. Contraction");
+    
+            copy_upper_lower_wAO_core_symm(M2p, wppq, begin, end);
+        } else {
+            timer_on("DFH: AO Construction");
+            compute_sparse_pQq_blocking_p_symm(start, stop, M1p, eri);
+            timer_off("DFH: AO Construction");
 
-        timer_on("DFH: wAO Copy");
-        copy_upper_lower_wAO_core_symm(M1p, wppq, begin, end);
-        timer_off("DFH: wAO Copy");
+            // contract half metric inverse
+            timer_on("DFH: AO-Met. Contraction");
+            contract_metric_AO_core_symm(M1p, ppq, metp, begin, end);
+            timer_off("DFH: AO-Met. Contraction");
+    
+            // contract full metric inverse computes  [J^{-1.0}](Q|mn)
+            timer_on("DFH: AO-Met. Contraction");
+            contract_metric_AO_core_symm(M1p, m1ppq, met1p, begin, end);
+            timer_off("DFH: AO-Met. Contraction");
```

Comment:
is there an adv to switching order of half and full in non-wcombine case?

Reply:
None that informed its change

Reply:
In fact, there's only one contraction in wcombine now, so that shouldn't matter at all! I forgot to delete the code I'd commented out.

Reply:
Perhaps switch half and full back then so it doesn't look like a meaningful change.

Reply:
I didn't read your comment carefully enough. my bad.

Reply:
One of my latest commits fixes this issue